cmd/clusterctl/pkg/client/init_test.go

```diff
@@ -254,6 +341,9 @@ func Test_clusterctlClient_Init(t *testing.T) {
 			if (err != nil) != tt.wantErr {
 				t.Fatalf("error = %v, wantErr %v", err, tt.wantErr)
 			}
+			if tt.wantErr {
+				return
+			}
```

Comment:
Does this change the behaviour of the tests slightly? If you wanted an error and got one, you would get to this and then exit before checking which providers were installed. I would assume in a failure scenario we should be able to predict which providers get installed (maybe always none?) so the rest of the test should be able to continue as normal without this short circuit I would have thought? Have I missed something?

Reply:
This does not impact the test cases currently implemented.
However, there is a TODO in the tracking list for some improvements in managing failure scenarios on init, and when this will be implemented, we should reconsider how to test that no provider is installed in case of error (your point above). 
For the time being, I prefer being consistent with other tests in the clusterctl code base.